pkg/queue/cluster_queue_impl_test.go

```diff
@@ -305,10 +289,10 @@ func TestClusterQueueImpl(t *testing.T) {
 			}
 
 			for _, w := range test.inadmissibleWorkloadsToRequeue {
-				cq.RequeueIfNotPresent(w, RequeueReasonNamespaceMismatch)
+				cq.requeueIfNotPresent(w, false)
 			}
 			for _, w := range test.admissibleWorkloadsToRequeue {
-				cq.RequeueIfNotPresent(w, RequeueReasonGeneric)
+				cq.requeueIfNotPresent(w, true)
```

Comment:
aren't we missing coverage for RequeueIfNotPresent by doing this? It would be good to test that we don't requeue immediately when `RequeueReasonNamespaceMismatch` is passed, may be a separate small test then?